add-met.prop: Fatty Acid Ester Addition

wtscott31 opened the pull request:
### Main improvements in this PR:

This is to update the model to include fatty acid esters produced during alcoholic fermentation in the cytoplasm and extracellular membrane via the protein medium-chain fatty acid ethyl ester synthase/esterase 2.

**I hereby confirm that I have:**

- [x] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---contributor) for running the model
- [x] Selected `devel` as a target branch (top left drop-down menu)
- [x] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about this PR

edkerk commented:
What is the status of this? The current code adds all reactants in one reaction, while these should probably be a number of different reactions.

BenjaSanchez commented:
@edkerk also not sure of this. @wtscott31 what reactions were you intending to add here? Could you write below the stoichiometry? Hopefully with references to some database where they come from.

wtscott31 commented:
@BenjaSanchez I proposed to add reactions, metabolites, and genes that were missing related to amino acid degradation, fatty acid degradation, and sulfur reduction pathways. Here are the reactions and related stoichiometry as well as associated genes : 

r_4629	s_0682[m] + s_4207[m]  <=> s_0532[m] + s_4208[m] 	YBR177C or YPL095C	-1000.00	1000.00	  0.00
r_4630	s_0682[m] + s_4211[m]  <=> s_0532[m] + s_4212[m] 	YBR177C or YPL095C	-1000.00	1000.00	  0.00
r_4631	s_0682[m] + s_4215[m]  <=> s_0532[m] + s_4216[m] 	YBR177C or YPL095C	-1000.00	1000.00	  0.00
r_4632	s_0376[m] + s_4219[m]  <=> s_0532[m] + s_4220[m] 	YGR177C or YOR377W or YGR015C	-1000.00	1000.00	  0.00
r_4633	s_0682[m] + s_4223[c]  <=> s_0532[m] + s_4224[m] 	YBR177C or YPL095C	-1000.00	1000.00	  0.00
r_4634	s_4208[m]  <=> s_4209[c] 		-1000.00	1000.00	  0.00
r_4635	s_4212[m]  <=> s_4213[c] 		-1000.00	1000.00	  0.00
r_4636	s_4216[m]  <=> s_4217[c] 		-1000.00	1000.00	  0.00
r_4637	s_4220[m]  -> s_4221[c] 		 -0.00	1000.00	  0.00
r_4638	s_4224[m]  -> s_4225[c] 		 -0.00	1000.00	  0.00
r_4639	s_4209[c]  <=> s_4210[e] 		-1000.00	1000.00	  0.00
r_4640	s_4213[c]  <=> s_4214[e] 		-1000.00	1000.00	  0.00
r_4641	s_4217[c]  <=> s_4218[e] 		-1000.00	1000.00	  0.00
r_4642	s_4221[c]  -> s_4222[e] 		 -0.00	1000.00	  0.00
r_4643	s_4225[c]  -> s_4226[e] 		 -0.00	1000.00	  0.00
r_4644	s_4210[e]  <=> 		-1000.00	1000.00	  0.00
r_4645	s_4214[e]  <=> 		-1000.00	1000.00	  0.00
r_4646	s_4218[e]  <=> 		-1000.00	1000.00	  0.00
r_4647	s_4222[e]  <=> 		-1000.00	1000.00	  0.00
r_4648	s_4226[e]  <=> 		-1000.00	1000.00	  0.00
r_4649	s_0376[m] + s_0682[m]  -> s_0532[m] + s_4227[m] 	YGR015C	 -0.00	1000.00	  0.00
r_4650	s_4227[m]  -> s_0685[c] 		 -0.00	1000.00	  0.00
r_4651	s_0178[c] + s_0794[c]  -> s_0456[c] + s_4228[c] 	YGR087C or YLR044C or YLR134W	 -0.00	1000.00	  0.00
r_4652	s_0794[c] + s_1203[c] + s_4228[c]  -> s_1198[c] + s_4231[c] 	YBR145W or YDL168W or YOL086C	 -0.00	1000.00	  0.00
r_4653	s_0799[m] + s_1205[m] + s_4229[m]  -> s_1200[m] + s_4232[m] 	YGL256W or YMR083W	 -0.00	1000.00	  0.00
r_4654	s_0794[c] + s_1212[c] + s_4228[c]  -> s_1207[c] + s_4231[c] 	YCR105W or YDR368W or YMR318C	 -0.00	1000.00	  0.00
r_4655	s_4231[c]  <=> s_4232[m] 		-1000.00	1000.00	  0.00
r_4656	s_4228[c]  <=> s_4230[e] 		-1000.00	1000.00	  0.00
r_4657	s_4231[c]  <=> s_4233[e] 		-1000.00	1000.00	  0.00
r_4658	s_4230[e]  -> 		 -0.00	1000.00	  0.00
r_4659	s_4233[e]  -> 		 -0.00	1000.00	  0.00
r_4660	s_0294[c] + s_0794[c]  -> s_0456[c] + s_4234[c] 	YHR137W	 -0.00	1000.00	  0.00
r_4661	s_0794[c] + s_1203[c] + s_4234[c]  -> s_1198[c] + s_4237[c] 	YBR145W or YDL168W or YOL086C	 -0.00	1000.00	  0.00
r_4662	s_0799[m] + s_1205[m] + s_4235[m]  -> s_1200[m] + s_4238[m] 	YGL256W or YMR083W	 -0.00	1000.00	  0.00
r_4663	s_0794[c] + s_1212[c] + s_4234[c]  -> s_1207[c] + s_4237[c] 	YCR105W or YDR368W or YMR318C	 -0.00	1000.00	  0.00
r_4664	s_4237[c]  <=> s_4238[m] 		-1000.00	1000.00	  0.00
r_4665	s_4234[c]  <=> s_4236[e] 		-1000.00	1000.00	  0.00
r_4666	s_4237[c]  <=> s_4239[e] 		-1000.00	1000.00	  0.00
r_4667	s_4236[e]  -> 		 -0.00	1000.00	  0.00
r_4668	s_4239[e]  -> 		 -0.00	1000.00	  0.00
r_4669	s_0204[c] + s_0794[c]  -> s_0456[c] + s_4240[c] 	YGR087C or YLR044C or YLR134W	 -0.00	1000.00	  0.00
r_4670	s_0794[c] + s_1203[c] + s_4240[c]  -> s_1198[c] + s_4243[c] 	YBR145W or YDL168W or YOL086C	 -0.00	1000.00	  0.00
r_4671	s_0799[m] + s_1205[m] + s_4241[m]  -> s_1200[m] + s_4244[m] 	YGL256W or YMR083W	 -0.00	1000.00	  0.00
r_4672	s_0794[c] + s_1212[c] + s_4240[c]  -> s_1207[c] + s_4243[c] 	YCR105W or YDR368W or YMR318C	 -0.00	1000.00	  0.00
r_4673	s_4243[c]  <=> s_4244[m] 		-1000.00	1000.00	  0.00
r_4674	s_4240[c]  <=> s_4242[e] 		-1000.00	1000.00	  0.00
r_4675	s_4243[c]  <=> s_4245[e] 		-1000.00	1000.00	  0.00
r_4676	s_4242[e]  -> 		 -0.00	1000.00	  0.00
r_4677	s_4245[e]  -> 		 -0.00	1000.00	  0.00
r_4678	s_0803[c] + s_1198[c] + s_4228[c]  <=> s_0794[c] + s_1203[c] + s_4246[c] 	YMR170C or YMR169C or YOR374W	-1000.00	1000.00	  0.00
r_4679	s_0439[p] + s_0534[p] + s_4247[p]  <=> s_0426[p] + s_0638[p] + s_4248[p] 	YOR317W or YER015W or YIL009W or YMR246W	-1000.00	1000.00	  0.00
r_4680	s_0682[m] + s_1382[m]  <=> s_0532[m] + s_4249[m] 	YBR177C or YPL095C	-1000.00	1000.00	  0.00
r_4681	s_4248[p]  <=> s_1382[m] 		-1000.00	1000.00	  0.00
r_4682	s_2884[p]  <=> s_4215[m] 		-1000.00	1000.00	  0.00
r_4683	s_2885[p]  <=> s_4207[m] 		-1000.00	1000.00	  0.00
r_4684	s_1258[p]  <=> s_4211[m] 		-1000.00	1000.00	  0.00
r_4685	s_0605[p]  <=> s_4223[c] 		-1000.00	1000.00	  0.00
r_4686	s_4246[c]  <=> s_4247[p] 		-1000.00	1000.00	  0.00
r_4687	s_0373[c] + s_4243[c]  -> s_0529[c] + s_4252[c] 	YGR177C or YOR377W or YGR015C	 -0.00	1000.00	  0.00
r_4688	s_4252[c]  -> s_4253[e] 		 -0.00	1000.00	  0.00
r_4689	s_4253[e]  -> 		 -0.00	1000.00	  0.00
r_4690	s_0373[c] + s_4237[c]  -> s_0529[c] + s_4254[c] 	YGR177C or YOR377W or YGR015C	 -0.00	1000.00	  0.00
r_4691	s_4254[c]  -> s_4255[e] 		 -0.00	1000.00	  0.00
r_4692	s_4255[e]  -> 		 -0.00	1000.00	  0.00
r_4693	s_0373[c] + s_4231[c]  -> s_0529[c] + s_4256[c] 	YGR177C or YOR377W or YGR015C	 -0.00	1000.00	  0.00
r_4694	s_4256[c]  -> s_4257[e] 		 -0.00	1000.00	  0.00
r_4695	s_4257[e]  -> 		 -0.00	1000.00	  0.00
r_4696	s_4234[c]  -> 2 s_0456[c] + s_1150[c] + s_4258[c] 		 -0.00	1000.00	  0.00
r_4697	s_1150[c] + s_4113[c]  <=> s_0841[c] + s_1469[c] + s_4259[c] 	YOR286W	-1000.00	1000.00	  0.00
r_4698	s_4249[m]  <=> s_4250[c] 		-1000.00	1000.00	  0.00
r_4699	s_4250[c]  <=> s_4251[e] 		-1000.00	1000.00	  0.00
r_4700	s_4251[e]  <=> 		-1000.00	1000.00	  0.00

feiranl commented:
Hi, could you print the reaction formula using the metname? printRxnFormula(model_original,'rxnAbbrList',[where you put your rxnID, eg: 'r_4700'],'metNameFlag',true);

wtscott31 commented:
OK, I can do that. Here they are as follows:

  | r_4629 | alcohol acyltransferase   (hexanoyl-CoA) | hexanoyl-CoA[m] + ethanol[m] <=>   ethyl hexanoate[m] + coenzyme A[m] | 2.3.1.84 | YBR177C or YPL095C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
  | r_4630 | alcohol acyltransferase (octanoyl-CoA) | octanoyl-CoA[m] + ethanol[m] <=> ethyl octanoate[m] + coenzyme A[m] | 2.3.1.84 | YBR177C or YPL095C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
  | r_4631 | alcohol acyltransferase (butyryl-CoA) | butanoyl-CoA[m] + ethanol[m] <=> ethyl butanoate[m] +   coenzyme A[m] | 2.3.1.84 | YBR177C or YPL095C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
  | r_4632 | alcohol acetyltransferase (hexanol) | acetyl-CoA[m] + hexanol[m] <=> hexyl acetate[m] +   coenzyme A[m] | 2.3.1.84 | YGR177C or YOR377W or YGR015C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
  | r_4633 | alcohol acyltransferase (decanoyl-CoA) | decanoyl-CoA[m] + ethanol[m] <=> ethyl decanoate[m] +   coenzyme A[m] | 2.3.1.84 | YBR177C or YPL095C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
  | r_4634 | ethyl hexanoate transport, mitochondrial | ethyl hexanoate[m] <=> ethyl hexanoate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4635 | ethyl octanoate transport, mitochondrial | ethyl octanoate[m] <=> ethyl octanoate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4636 | ethyl butanoate transport, mitochondrial | ethyl butanoate[m] <=> ethyl butanoate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4637 | hexyl acetate transport, mitochondrial | hexyl acetate[m] => hexyl acetate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4638 | ethyl decanoate transport, mitochondrial | ethyl decanoate[m] => ethyl decanoate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4639 | ethyl hexanoate transport | ethyl hexanoate[c] <=> ethyl hexanoate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4640 | ethyl octanoate transport | ethyl octanoate[c] <=> ethyl octanoate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4641 | ethyl butanoate transport | ethyl butanoate[c] <=> ethyl butanoate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4642 | hexyl acetate transport | hexyl acetate[c] => hexyl acetate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4643 | ethyl decanoate transport | ethyl decanoate[c] => ethyl decanoate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4644 | ethyl hexanoate exchange | ethyl hexanoate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4645 | ethyl octanoate exchange | ethyl octanoate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4646 | ethyl butanoate exchange | ethyl butanoate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4647 | hexyl acetate exchange | hexyl acetate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4648 | ethyl decanoate exchange | ethyl decanoate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4649 | mitochondrial ethanol O-acetyltransferase | ethanol[m] + acetyl-CoA[m] => ethyl acetate[m] + coenzyme   A[m] | 2.3.1.268 | YGR015C |   |   |   |   | kegg.reaction/R11957;sbo/SBO:0000176 |   |   |   | 2
  | r_4650 | ethyl acetate transport, mitochondrial | ethyl acetate[m] => ethyl acetate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4651 | pyruvate decarboxylase (aldedyde-forming) | H+[c] + 2-oxobutanoate[c] => carbon dioxide[c] +   propanal[c] | 4.1.1.1 | YGR087C or YLR044C or YLR134W |   |   |   |   | kegg.reaction/R00636;sbo/SBO:0000176 |   |   |   | 2
  | r_4652 | aldehyde dehydrogenase (1-propanol, NAD) | propanal[c] + H+[c] + NADH[c] => propanol[c] + NAD[c] | 1.1.1.-; 1.1.1.1; 1.1.1.284 | YBR145W or YDL168W or YOL086C |   |   |   |   | kegg.reaction/R00754;metanetx.reaction/MNXR95725;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4653 | aldehyde dehydrogenase (1-propanol, NAD) | propanal[m] + H+[m] + NADH[m] => propanol[m] + NAD[m] | 1.1.1.1 | YGL256W or YMR083W |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12702265;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics;sce01200  Carbon metabolism | 3
  | r_4654 | aldehyde dehydrogenase (1-propanol, NADP) | propanal[c] + H+[c] + NADPH[c] => propanol[c] + NADP(+)[c] | 1.1.1.-; 1.1.1.2 | YCR105W or YDR368W or YMR318C |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4655 | 1-propyl alcohol transport, mitochondrial | propanol[c] <=> propanol[m] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4656 | propanal transport, cytosol | propanal[c] <=> propanal[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4657 | 1-propyl alcohol transport, cytosol | propanol[c] <=> propanol[e] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4658 | propanal exchange | propanal[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4659 | propanol exchange | propanol[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4660 | 2-oxo acid   decarboxylase | H+[c] + 4-methylthio-2-oxobutanoate[c] => carbon dioxide[c]   + methional[c] | 4.1.1.1 | YHR137W |   |   |   |   | kegg.reaction/R00636;sbo/SBO:0000176 |   |   |   | 0
  | r_4661 | aldehyde dehydrogenase (methionol, NAD) | methional[c] + H+[c] + NADH[c] => methionol[c] + NAD[c] | 1.1.1.-; 1.1.1.1; 1.1.1.284 | YBR145W or YDL168W or YOL086C |   |   |   |   | kegg.reaction/R00754;metanetx.reaction/MNXR95725;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4662 | aldehyde dehydrogenase (methionol, NAD) | methional[m] + H+[m] + NADH[m] => methionol[m] + NAD[m] | 1.1.1.1 | YGL256W or YMR083W |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12702265;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics;sce01200  Carbon metabolism | 3
  | r_4663 | aldehyde dehydrogenase (methionol, NADP) | methional[c] + H+[c] + NADPH[c] => methionol[c] +   NADP(+)[c] | 1.1.1.-; 1.1.1.2 | YCR105W or YDR368W or YMR318C |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4664 | methionol transport, mitochondrial | methionol[c] <=> methionol[m] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4665 | methional transport, cytosol | methional[c] <=> methional[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4666 | methionol transport, cytosol | methionol[c] <=> methionol[e] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4667 | methional exchange | methional[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4668 | methionol exchange | methionol[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4669 | pyruvate decarboxylase (hydroxy-phenyl) | H+[c] + 3-(4-hydroxyphenyl)pyruvate[c] => carbon dioxide[c]   + (4-hydroxyphenyl)acetaldehyde[c] | 4.1.1.1 | YGR087C or YLR044C or YLR134W |   |   |   |   | kegg.reaction/R00636;sbo/SBO:0000655 |   |   |   | 2
  | r_4670 | aldehyde dehydrogenase (tyrosol, NAD) | (4-hydroxyphenyl)acetaldehyde[c] + H+[c] + NADH[c] =>   tyrosol[c] + NAD[c] | 1.1.1.-; 1.1.1.1; 1.1.1.284 | YBR145W or YDL168W or YOL086C |   |   |   |   | kegg.reaction/R00754;metanetx.reaction/MNXR95725;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4671 | aldehyde dehydrogenase (tyrosol, NAD) | (4-hydroxyphenyl)acetaldehyde[m] + H+[m] + NADH[m] =>   tyrosol[m] + NAD[m] | 1.1.1.1 | YGL256W or YMR083W |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12702265;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics;sce01200  Carbon metabolism | 3
  | r_4672 | aldehyde dehydrogenase (tyrosol, NADP) | (4-hydroxyphenyl)acetaldehyde[c] + H+[c] + NADPH[c] =>   tyrosol[c] + NADP(+)[c] | 1.1.1.-; 1.1.1.2 | YCR105W or YDR368W or YMR318C |   |   |   |   | metanetx.reaction/MNXR95716;pmid/12499363;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics | 3
  | r_4673 | tyrosol transport, mitochondrial | tyrosol[c] <=> tyrosol[m] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4674 | (4-hydroxyphenyl)acetaldehyde transport, cytosol | (4-hydroxyphenyl)acetaldehyde[c] <=>   (4-hydroxyphenyl)acetaldehyde[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4675 | tyrosol transport, cytosol | tyrosol[c] <=> tyrosol[e] |   |   |   |   |   |   | metanetx.reaction/MNXR100780;sbo/SBO:0000655 |   |   |   | 0
  | r_4676 | (4-hydroxyphenyl)acetaldehyde exchange | (4-hydroxyphenyl)acetaldehyde[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4677 | tyrosol exchange | tyrosol[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 0
  | r_4678 | aldehyde dehydrogenase | propanal[c] + NAD[c] + H2O[c] <=> propionate[c] + H+[c]   + NADH[c] | 1.2.1.5 | YMR170C or YMR169C or YOR374W |   |   |   |   | kegg.reaction/R00538 ;sbo/SBO:0000176 | Gluconeogenesis;sce00010  Glycolysis;sce00071  Fatty acid degradation;sce00350  Tyrosine metabolism;sce01110  Biosynthesis of secondary   metabolites;sce01130  Biosynthesis of   antibiotics |   |   |   | 3
  | r_4679 | short-chain-fatty-acid-CoA ligase (propionate) | ATP[p] + coenzyme A[p] + propionate[p] <=> AMP[p] +   propionyl-CoA[p] + diphosphate[p] | 6.2.1.3 | YOR317W or YER015W or   YIL009W or YMR246W |   |   |   | kegg.reaction/R00389 ;sbo/SBO:0000176 | sce00061  Fatty acid biosynthesis;sce00071  Fatty acid degradation;sce01212  Fatty acid metabolism;sce04146  Peroxisome | 3
  | r_4680 | alcohol acyltransferase (propionyl-CoA) | propionyl-CoA[m] + ethanol[m] <=> ethyl propionate[m] +   coenzyme A[m] | 2.3.1.84 | YBR177C or YPL095C |   |   |   |   | kegg.reaction/R00627;metanetx.reaction/MNXR125910;sbo/SBO:0000655 |   |   | 2
  | r_4681 | propionyl-CoA transport, mitochondrial | propionyl-CoA[p] <=> propionyl-CoA[m] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4682 | butyryl-CoA transport, mitochondrial | butanoyl-CoA[p] <=> butanoyl-CoA[m] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4683 | hexanoyl-CoA transport, mitochondrial | hexanoyl-CoA[p] <=> hexanoyl-CoA[m] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4684 | octanoyl-CoA transport, mitochondrial | octanoyl-CoA[p] <=> octanoyl-CoA[m] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4685 | decanoyl-CoA transport, mitochondrial | decanoyl-CoA[p] <=> decanoyl-CoA[m] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4686 | propionate transport, cytosol | propionate[c] <=> propionate[p] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 0
  | r_4687 | alcohol acetyltransferase (tyrosol) | acetyl-CoA[c] + tyrosol[c] => coenzyme A[c] + tyrosyl   acetate[c] | 2.3.1.84 | YGR177C or YOR377W or YGR015C |   |   |   |   | metanetx.reaction/MNXR102164;pmid/12957907;sbo/SBO:0000176 |   |   |   | 3
  | r_4688 | tyrosyl acetate transport | tyrosyl acetate[c] => tyrosyl acetate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4689 | tyrosyl acetate exchange | tyrosyl acetate[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4690 | alcohol acetyltransferase (methionol) | acetyl-CoA[c] + methionol[c] => coenzyme A[c] + methionyl   acetate[c] | 2.3.1.84 | YGR177C or YOR377W or YGR015C |   |   |   |   | metanetx.reaction/MNXR102164;pmid/12957907;sbo/SBO:0000176 |   |   |   | 3
  | r_4691 | methionyl acetate transport | methionyl acetate[c] => methionyl acetate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4692 | methionyl acetate exchange | methionyl acetate[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4693 | alcohol acetyltransferase (propanol) | acetyl-CoA[c] + propanol[c] => coenzyme A[c] + propyl   acetate[c] | 2.3.1.84 | YGR177C or YOR377W or YGR015C |   |   |   |   | metanetx.reaction/MNXR102164;pmid/12957907;sbo/SBO:0000176 |   |   |   | 3
  | r_4694 | propyl acetate transport | propyl acetate[c] => propyl acetate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4695 | propyl acetate exchange | propyl acetate[e] => |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4696 | methanethiol formation | methional[c] => ethylene[c] + methanethiol[c] + 2 carbon   dioxide[c] |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1
  | r_4697 | thiosulfate—thiol sulfurtransferase | methanethiol[c] + thiosulfate[c] <=> sulphite[c] +   hydrogen sulfide[c] + dimethyl disulfide[c] | 2.8.1.3 | YOR286W |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 2
  | r_4698 | ethyl propionate transport, mitochondrial | ethyl propionate[m] <=> ethyl propionate[c] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4699 | ethyl propionate transport | ethyl propionate[c] <=> ethyl propionate[e] |   |   |   |   |   |   | sbo/SBO:0000655 |   |   |   |   | 1
  | r_4700 | ethyl propionate exchange | ethyl propionate[e] <=> |   |   |   |   |   |   | sbo/SBO:0000627 |   |   |   |   | 1

 

wtscott31 commented:
However, I now remember why I only  have one reaction when I submitted that PR request. It was because in  MetaNetX there is only one generic reaction for alcohol O-acetyltransferase. Also, a lot of the reactions I added do not seem to have a MetaNetX identifier. What should I do then? I have all of my additions in a xml and xlsx file merged with yeastGEM though.

edkerk commented:
@wtscott31 You probably modified a script that added reactions based on MetaNetX IDs, which is not so suitable for your case. But considering that you seem to have all the necessary information already, you could quite easily include the new metabolites and reactions that you propose with just a couple of lines. Use the `addMultipleMetabolites` and ´addMultipleReactions` functions from COBRA toolbox, and provide the necessary information by loading a properly formatted TSV file. You can also have a look at an example of this using the RAVEN toolbox in [this repo](https://github.com/SysBioChalmers/rhto-GEM/blob/master/ComplementaryScripts/reconstruction/r3_add_rhto_specific_reactions.m), line 13-41, but note that the function names and input format is a little different when using RAVEN. Both approaches work fine if performed correctly :)

wtscott31 commented:
Thanks @edkerk for your suggestions. I have modified the script accordingly and I am ready to add it to this PR, but I cannot push the commit through Github for some reason.

edkerk reviewed (changes requested):
To solve the merge conflict, could you copy-paste the latest XML model file from the `devel` branch: https://github.com/SysBioChalmers/yeast-GEM/blob/devel/ModelFiles/xml/yeastGEM.xml, into your local yeast-GEM repository, re-run the whole script that adds these new reactions, and then make a new commit with these new model files?

Also, some old files can be deleted.

wtscott31 commented:
> To solve the merge conflict, could you copy-paste the latest XML model file from the `devel` branch: https://github.com/SysBioChalmers/yeast-GEM/blob/devel/ModelFiles/xml/yeastGEM.xml, into your local yeast-GEM repository, re-run the whole script that adds these new reactions, and then make a new commit with these new model files?
> 
> Also, some old files can be deleted.

Hi @edkerk I made those changes you recommended in the aforementioned statement. I hope that this works. Let me know if this is fine. If not I can re-run the model or try something else.

edkerk commented:
I think you might have copy-pasted the `yeastGEM.xml` from the `devel` branch on your local computer, **without** first fetching the lastest updates on that branch. It seems like your local `devel` branch did not yet incorporate changes from e.g. #228, as for instance reaction `r_0073` is still irreversible, while the latest model in `devel` has this reaction set as reversible.

To ensure that you have the correct latest `yeastGEM.xml` file from the `devel` branch, it's probably easiest to download the model directly from this link: https://github.com/SysBioChalmers/yeast-GEM/raw/devel/ModelFiles/xml/yeastGEM.xml (right click, "Save as..."), or [this page](https://github.com/SysBioChalmers/yeast-GEM/blob/devel/ModelFiles/xml/yeastGEM.xml), and then re-run your code again.

Without this, GitHub is not able to properly identify the changes in the model files and this is preventing us to accept the changes. Thanks for your efforts!

wtscott31 commented:
Hi @edkerk, I ran the code again using yeastGEM from the link you sent. I am also sending you the getNewIndex function
[getNewIndex.zip](https://github.com/SysBioChalmers/yeast-GEM/files/5158215/getNewIndex.zip)

edkerk commented:
Not sure why it's still reporting a conflict. Could you please check whether "Allow edits from maintainers" is enabled in this pull request? (explanation: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)

wtscott31 commented:
Yes,  the "allow edits from maintainers" box is checked.

wtscott31 commented:
> Not sure why it's still reporting a conflict. Could you please check whether "Allow edits from maintainers" is enabled in this pull request? (explanation: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)

The changes work locally. Should I revert to using another version of yeastGEM?

edkerk commented:
I'm trying to sort it out here, but I'm still getting stuck. The solution I'm trying to follow is to first merge the `devel` branch into the branch on your fork, but this is also not without issues.

Perhaps the easiest solution is this:

1. Make a new `devel` branch in [your fork on GitHub](https://github.com/wtscott31/yeast-GEM/). 
2. Afterwards, make a pull request from the original repository to your fork, using [this link](https://github.com/wtscott31/yeast-GEM/compare/devel...SysBioChalmers:devel) (click Create Pull Request; and then Merge pull request, only works if you have first made a `devel` branch on your fork).
3. While in the [`devel` branch on your fork](https://github.com/wtscott31/yeast-GEM/compare/devel...SysBioChalmers:devel), make a new branch for the FAE additions.
4. Copy-paste the necessary code and .tsv files to your new branch.
5. Push the changes to GitHub and make a new pull request towards the `devel` branch on the original yeast-GEM repository.

This should work as it basically starts with a clean sheet.

edkerk commented:
@wtscott31 Ignore my previous comment, I succeeded!

wtscott31 commented:
ok great! Thanks for your help @edkerk and @BenjaSanchez!

BenjaSanchez commented on ComplementaryScripts/modelCuration/addFAEnewRxn.m:
```diff
@@ -0,0 +1,162 @@
+        model.metCharges(metID)  =  newmet.metCharges(i);
+        model.metKEGGID{metID}   = newmet.metKEGGID{i};
+        model.metChEBIID{metID}  = newmet.metChEBIID{i};
+        model.metNotes{metID}    = ['NOTES: added after new annotation (PR #142); ',newmet.metNotes{i}];
```
This line should refer to the current PR (#190)

wtscott31 replied:
ok I can change it

BenjaSanchez commented on ComplementaryData/modelCuration/FAEnewRxnMetAnnotation.tsv:
```diff
@@ -0,0 +1,53 @@
+NewMetName	Charged formula 	Charge	compartment	KEGG ID	CHEBI ID	Remark
+ethyl hexanoate[m]	C8H16O2	0	mitochondrion		CHEBI:86055	
+ethyl hexanoate[c]	C8H16O2	0	cytoplasm		CHEBI:86055	
+ethyl hexanoate[e]	C8H16O2	0	endoplasmic reticulum		CHEBI:86055	
```
Here you used `e` for endoplasmic reticulum but it should have been `er`, as `e` is reserved for "extracellular".

wtscott31 replied:
Thanks @BenjaSanchez for catching that, but the [e] is correct. However, extracellular should be used instead of endoplasmic reticulum.

BenjaSanchez reviewed (commented):
@wtscott31 @edkerk apologies for being too late for the review, was going through the PR only today. I see that the following metabolite/reaction fields were not added to the model, although they are present in the data files:

* Metabolites: `metFormulas`, `metCharges`, `metChEBIID` and `metKEGGID` ids are all missing, as they don't show up in the `.xml`.
* Reactions: KEGG pathways in the `subSystems` field are also missing.

There're a couple more issues below, but the main problem at the moment is that the model won't open in Matlab, not really sure why (looks like due to some missing field). I get the following error when using `loadYeastModel` from devel, @edkerk do you see the same behavior in your setup?

![image](https://user-images.githubusercontent.com/9384349/91996381-3537f580-ed39-11ea-9058-b14217cf1371.png)

Not sure what should be done here @edkerk should a new PR be opened addressing these issues? Or instead should this PR first be reverted, and on top of that branch add the metabolite/reactions again, this time isolating the error? I'm aiming towards the second option, as in it's current state the model cannot be opened with COBRA for correcting the problem.

wtscott31 commented:
> @wtscott31 @edkerk apologies for being too late for the review, was going through the PR only today. I see that the following metabolite/reaction fields were not added to the model, although they are present in the data files:
> 
> * Metabolites: `metFormulas`, `metCharges`, `metChEBIID` and `metKEGGID` ids are all missing, as they don't show up in the `.xml`.
> * Reactions: KEGG pathways in the `subSystems` field are also missing.
> 
> There're a couple more issues below, but the main problem at the moment is that the model won't open in Matlab, not really sure why (looks like due to some missing field). I get the following error when using `loadYeastModel` from devel, @edkerk do you see the same behavior in your setup?
> 
> ![image](https://user-images.githubusercontent.com/9384349/91996381-3537f580-ed39-11ea-9058-b14217cf1371.png)
> 
> Not sure what should be done here @edkerk should a new PR be opened addressing these issues? Or instead should this PR first be reverted, and on top of that branch add the metabolite/reactions again, this time isolating the error? I'm aiming towards the second option, as in it's current state the model cannot be opened with COBRA for correcting the problem.

How should I address this?

edkerk commented:
@BenjaSanchez I get the same error, weird because it didn't complain when running the code. I suppose I got too excited about getting the merge conflict solved. Let's revert this PR for now, to have a functional model in `devel`, and make a new branch to re-apply the code.

Not sure what is causing the problem, it also fails when loading with RAVEN, as the issue is that `translateSBML` does not return a model structure. RAVEN shows that `translateSBML` errors and states that the model is not valid SBML, but running it through the online SBML validator only returns a warning that metabolite SBO term 0000649 is not in full compliance. But that can't be a reason for not returning a model, also because these SBO terms did not change in this PR.

edkerk commented:
@BenjaSanchez @wtscott31 Found it, the `—` in the reaction name of `r_4697`

Run in the `ModelFiles\xml` folder:
`[~, error] = TranslateSBML('yeastGEM.xml',1,1,[1 1]);` 

Output:
```matlab
  struct with fields:

        line: 205203
     errorId: 1005
    severity: 'Error'
     message: 'Invalid character in XML content.↵'
```

BenjaSanchez commented:
@edkerk great find, thanks. @wtscott31 probably the easiest is that I open a new PR with all these changes and then you can go through it to see if everything looks good.

BenjaSanchez commented:
I opened #248 for addressing all remaining issues, @wtscott31 please let me know there if I forgot anything
